src/ssh/azext_ssh/target_properties_utils.py

```diff
@@ -0,0 +1,142 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import colorama
+
+from azure.cli.core import telemetry
+from azure.cli.core import azclierror
+from knack import log
+from . import constants as const
+
+logger = log.get_logger(__name__)
+
+# Send target OS type telemetry and check if authentication options are valid for that OS.
+def handle_target_machine_properties(cmd, op_info):
+
+    properties = get_properties(cmd, op_info.resource_type.lower(), op_info.resource_group_name, op_info.vm_name)
+    
+    if properties:
+        os_type = parse_os_type(properties, op_info.resource_type.lower())
+        agent_version = parse_agent_version(properties, op_info.resource_type.lower())
+        tags = parse_tags(properties, op_info.resource_type.lower())
+    else:
+        os_type, agent_version, tags = None, None, None
+
+    check_valid_os_type(os_type, agent_version, op_info)
+
+    op_info.set_azure_resource_tags(tags)
```

Comment:
Maybe create a function just for doing everything related to handling the resource tags and setting the port.